images/virtualization-artifact/pkg/controller/cvi/internal/source/http.go

```diff
@@ -59,8 +60,9 @@ func NewHTTPDataSource(
 func (ds HTTPDataSource) Sync(ctx context.Context, cvi *virtv2.ClusterVirtualImage) (reconcile.Result, error) {
 	log, ctx := logger.GetDataSourceContext(ctx, "http")
 
-	condition, _ := service.GetCondition(cvicondition.ReadyType, cvi.Status.Conditions)
-	defer func() { service.SetCondition(condition, &cvi.Status.Conditions) }()
+	condition, _ := conditions.GetCondition(cvicondition.ReadyType, cvi.Status.Conditions)
+	cb := conditions.NewConditionBuilder(cvicondition.ReadyType).FromCondition(condition).Generation(cvi.Generation)
+	defer func() { conditions.SetCondition(cb, &cvi.Status.Conditions) }()
```

Comment:
Will result in an error if reason is not set

Reply:
It's logic problem, not builder